packages/components/src/confirm/component.tsx

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type { Ref } from 'react';
+import { Card, CardHeader, CardFooter } from '../card';
+import { Button } from '../button';
+import { Heading } from '../heading';
+
+/**
+ * Internal dependencies
+ */
+import type { OwnProps } from './types';
+import { useConfirm } from './hook';
+import { contextConnect, PolymorphicComponentProps } from '../ui/context';
+
+function Confirm(
+	props: PolymorphicComponentProps< OwnProps, 'div' >,
+	forwardedRef: Ref< any >
+) {
+	const { role, wrapperClassName, ...otherProps } = useConfirm( props );
+
+	return (
+		<div { ...otherProps } role={ role } className={ wrapperClassName }>
+			<Card ref={ forwardedRef }>
```

Comment:
I thought the issue was related to not passing the `forwardedRef` correctly, so I made a few trial and errors attempts. There's no specific reason this is set in this component. In fact, AFAIK, this component doesn't need to expose or forward a ref to a child, but I'm probably missing an aspect of the G2 API that requires refs to be exposed and forwarded like that? 🤔

Reply:
The context system does require/expect that refs are forwarded for all components. We typically make the ref on the outermost wrapper component and make sure to use `View` instead of `div` for the wrapper so that the `as` prop works. If you don't want polymorphism for this component then make sure to pass `false` to PolymorphicComponentProps as the third type parameter to denote that it is not Polymorphic (please excuse the confusing type name, it'll be updated to WordPressComponent soon).

Reply:
@ciampo For when you have more time to tinker with the context system, it might be worth adding the ability to _not_ forward refs for a component. I think this could be accomplished by first checking the `length` property on the component function that is passed to `contextConnect` before passing it to `forwardRef` inside of `contextConnect`. Not sure if that's worth it but might be for some components where it doesn't make sense to expose a ref (if indeed it makes sense for some to not do so, I'm undecided one that).